src/games/strategy/triplea/ui/RouteOptimizer.java

```diff
@@ -22,60 +22,115 @@
   private static final int commonAdditionalScreens = 2;
   private static final int minAdditionalScreens = 0;
 
-  private final double gameMapWidth;
+  private Point endPoint;
   private int mapWidth;
   private int mapHeight;
 
   public RouteOptimizer(MapData mapData, MapPanel mapPanel) {
     checkNotNull(mapData);
     this.mapPanel = checkNotNull(mapPanel);
-    mapWidth = mapPanel.getImageWidth();
-    mapHeight = mapPanel.getImageHeight();
-
     isInfiniteY = mapData.scrollWrapY();
-
     isInfiniteX = mapData.scrollWrapX();
-
-    gameMapWidth = mapData.getMapDimensions().getWidth();
-
   }
 
-  List<Point> normalizeForHorizontalWrapping(Point... route) {
-    final boolean crossesMapSeam = crossMapSeam(route);
-    final List<Point> points = Arrays.asList(route);
-
-    if (crossesMapSeam) {
-      final int minX = minX(route);
-      for (Point p : points) {
-        if ((p.x - minX) > mapPanel.getWidth()) {
-          p.x -= this.gameMapWidth;
+  /**
+   * Algorithm for finding the shortest path for the given Route
+   * 
+   * @param route The joints on the Map
+   * @return A Point array which goes through Map Borders if necessary
+   */
+  public Point[] getTranslatedRoute(Point... route) {
```

Comment:
We want to release 1.9 right? Since everything is working fine now, could we just release 1.9 and create the test cases afterwards? I'll promise you, that I will add test case right after I finished migrating to javafx!